benchmarks/timeseries/benchmark_datasourcing.py

```diff
@@ -0,0 +1,170 @@
+        "DataSourcingActor Request Channel"
+    )
+
+    channel_registry = ChannelRegistry(name="Microgrid Channel Registry")
+    request_receiver = request_channel.new_receiver(
+        "datasourcing-benchmark", maxsize=(num_evc_chargers * len(COMPONENT_METRIC_IDS))
+    )
+    request_sender = request_channel.new_sender()
+
+    consume_tasks = []
+
+    start_time = perf_counter()
+    samples_sent = 0
+
+    for evc_id in mock_grid.evc_ids:
+        for component_metric_id in COMPONENT_METRIC_IDS:
+            request = ComponentMetricRequest(
+                "current_phase_requests", evc_id, component_metric_id, None
+            )
+
+            recv_channel = channel_registry.new_receiver(request.get_channel_name())
+
+            async def consume(channel: Receiver[Any]) -> None:
+                while True:
+                    try:
+                        await channel.ready()
+                        channel.consume()
+                    except ReceiverStoppedError:
+                        return
+
+                    nonlocal samples_sent
+                    samples_sent += 1
+
+            await request_sender.send(request)
+            consume_tasks.append(asyncio.create_task(consume(recv_channel)))
+
+    DataSourcingActor(request_receiver, channel_registry)
+
+    await asyncio.gather(*consume_tasks)
+
+    time_taken = perf_counter() - start_time
+
+    await mock_grid.cleanup()
+
+    print(f"Samples Sent: {samples_sent}, time taken: {time_taken}")
+    print(f"Samples per second: {samples_sent / time_taken}")
+    print(
+        "Expected samples: "
+        f"{num_expected_messages}, missing: {num_expected_messages - samples_sent}"
+    )
+    print(
+        f"Missing per EVC: {(num_expected_messages - samples_sent) / num_evc_chargers}"
+    )
+
+
+def parse_args() -> Tuple[int, int, bool]:
+    """Parse the command line arguments.
+
+    Returns:
+        A tuple of (num evc chargers, num messages per battery, record allocations).
+    """
+    parser = argparse.ArgumentParser(description="Benchmark the data sourcing actor.")
+    parser.add_argument(
+        "num_evc_chargers",
+        type=int,
+        help="Number of EV Chargers to create for the mock microgrid.",
+    )
+    parser.add_argument(
+        "num_msgs_per_battery",
+        type=int,
+        help="Number of messages to send out for each battery.",
+    )
+    parser.add_argument(
+        "--record-allocations",
+        action="store_true",
+        help="Record memory allocations.",
+    )
+
+    args = parser.parse_args()
+
+    return args.num_evc_chargers, args.num_msgs_per_battery, args.record_allocations
+
+
+if __name__ == "__main__":
+    (
+        num_evc_chargers_pararm,
+        num_msgs_per_battery_param,
+        record_allocations,
+    ) = parse_args()
+
+    if record_allocations:
+        tracemalloc.start()
+
+    asyncio.run(
+        benchmark_data_sourcing(num_evc_chargers_pararm, num_msgs_per_battery_param)
+    )
```

Comment:
This is usually not a good practice to run programs/applications
https://github.com/frequenz-io/rnd/blob/master/doc/coding-standards/python.md#defining-applications